### ISSUE TITLE
 Bug module Blurs with Opencl - Driver AMD Radeon Vega Mobile Gfx on Windows 11 pro - Darktable 4.3

### ISSUE THREAD

**Issue by Christian-Bouhon** [darktable-org/darktable#14621]
### Describe the bug

Hello,
When I used the "Blurs" module, I couldn't export the image, it was either black or very dark and then I had to restart darktable.
If I disable opencl, no more problems apart from a longer execution time.
If I used the "Soften" module instead of the blursmodule, the problem was also solved.

Here is a link where you can find all the files. https://drive.google.com/drive/folders/10GVxUTK2e-SiGZM_vRgZflfBnGALlSt8?usp=sharing

![2021-07-03-20210703_0068- Blurs_LR](https://github.com/darktable-org/darktable/assets/120166629/5446f643-ead0-4c59-9406-8d68d8e80d04)
![2021-07-03-20210703_0068- Blurs without opencl_LR](https://github.com/darktable-org/darktable/assets/120166629/cd2c0b60-bb5e-4259-af77-6b95adc174e4)

### Steps to reproduce

I suggest you try out the "Blur" module and Opencl.
Here is a link where you can find all the files. https://drive.google.com/drive/folders/10GVxUTK2e-SiGZM_vRgZflfBnGALlSt8?usp=sharing

### Where did you install darktable from?

darktable.org

### darktable version

4.3.0+2427~g8c49e89404

### What OS are you using?

Windows

### What is the version of your OS?

Windows 11

### Describe your system?

Laptop Lenovo: AMD Ryzen 5 3500U with Radeon Vega Mobile Gfx

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

Laptop : AMD Ryzen 5 3500U with Radeon Vega Mobile Gfx

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

Too big for github

**Comment by jenshannoschwalm** [comment]
Checked on fedora linux and nvidia, no problem at all.

I guess your log with `-d opencl` doesn't show an error while processing blur?

Also we need the log for opencl device reports. Do you have any special setting in opencl preferences? Tried system resources setting to default?

**Comment by MStraeten** [comment]
I could see an issue with the old ‚soften‘ module with an improper rendering of the mask when used with opencl on my m1mac -even the image doesn’t turn black.
The ‚blur‘ module is fine

**Comment by Christian-Bouhon** [comment]
Hello,

I tried again this morning, but unfortunately I still have the same problem. Here are the log files. 
For the photo above, I had set the opencl to default and for this one to GPU veryfast.
I have to admit that this GPU isn't the best, I've never been able to configure it properly on Linux, which is why I have to use Windows (Pro 22h2 22621.1702) on this machine.

Thank you for your help,
Greetings from Brussels,
Christian
![2021-07-03-20210703_0068- Blurs](https://github.com/darktable-org/darktable/assets/120166629/c8f435e9-0759-456f-8e37-869227243631)
[darktable-log-old.txt](https://github.com/darktable-org/darktable/files/11624112/darktable-log-old.txt)
[darktable-log.txt](https://github.com/darktable-org/darktable/files/11624114/darktable-log.txt)

**Comment by jenshannoschwalm** [comment]
At least the first log is from dt 4.1?

I don't know what is wrong with that card on Linux but that platform is much better tested and if something is wrong on Linux I wouldn't even dare to test on Windows.

@piratenpanda could you check on your amd graphics?

**Comment by jenshannoschwalm** [comment]
Well, if you want someone to have a chance finding out a genuine dt bug we certainly require better and more precise information. A report from a 4.1 version is certainly not enough, there have been so many things fixed and it means we/i would have to look up the exact source code of that version... no chance to find someone doing so.

Best would be to compile dt yourself to get an understanding. Or you should at least note what version you precisely use.

And - there have been so many problems with amd drivers for some people. I am simply glad i chose an nvidia card ignoring being "closed source" in favour of stability and support :-)

So we must at least be sure everything has been set up correctly. I tested on nvidia and even on my intel notebook, no problems at all so it seems to be an issue of your system/driver

I would suggest this strategy
1. Update the title of the issue, mention amd driver ...
2. Use current dt (master from to today or wait for the 4.4 release version)
3. try to get OpenCL correctly running on linux, use clinfo and share the output
4. If that is all ok, try dt on your images with a log report of the "crashing" session

**Comment by Christian-Bouhon** [comment]
Hello,
Thank you very much for your reply. Here are the answers to your various points.
Sorry if I'm not precise enough, I'm not a computer development pro. FYI, I've spent my entire career in the bakery pastry sector (R&d and commercial management).


> 1. Update the title of the issue, mention amd driver ...

Done


> 2\. Use current dt (master from to today or wait for the 4.4 release version)

The tests were carried out on the latest master version : https://discuss.pixls.us/t/darktable-windows-insider-program-5-28/37277


> 3\. try to get OpenCL correctly running on linux, use clinfo and share the output

I persevered for over a year with no result, my desktop runs on Debian without any problem and "without Windows ;-) ".
I have also participated in several posts on pixl.us and darktable.fr without result. 
eg. https://discuss.pixls.us/t/opencl-amd-vega-8/32788

> 4\. If that is all ok, try dt on your images with a log report of the "crashing" session

Sorry for the confusion, I thought the old log file might be useful. The second one only contains the crash session(below).
[darktable-log.txt](https://github.com/darktable-org/darktable/files/11624114/darktable-log.txt)

Greetings from Brussels,
Christian

**Comment by jenshannoschwalm** [comment]
1. So at least amd improved the driver as your cards now fulfills the requirement of "image supported".
2. In your old discussion the clinfo had some concerning infos, it might be you'd have to switch off the graphics-card compiler options in your config file (make that line empty) but that's just a very vague idea. (you's have to delete the cached kernels ...)
3. While looking into blurs opencl code there are no bells ringing here.

Basically don't know whats wrong on your system.

**Comment by Christian-Bouhon** [comment]
Hello,
The problem is only with on my AMD Radeon Vega 8 Mobile Gfx video card under Windows 11. We have also done a test on my son's laptop (Intel i9 + Nvidia quadro T1000) with the same version of Windows and Darktable, and we have no problem ([darktable-log.txt](https://github.com/darktable-org/darktable/files/11639188/darktable-log.txt)).
I will try to compile Darktable on my Windows machine to get an optimised version for my AMD card. I won't promise this for next week as I'm under time pressure, I'm planning it for mid-June.
I suggest you close this post.
Greetings from Brussels,
Christian

**Comment by jenshannoschwalm** [comment]
Christian, you seem to be a bit unfortunate with your card :-) Plus - there are no dt devs developing on windows plus having an amd opencl device so we are "lost in the dark". If you stumble into something new let us know, probably a log with `-d pipe -d opencl` using 4.4 would be just right.

**Comment by Christian-Bouhon** [comment]
> If you stumble into something new let us know,

Hallo,
The problem has been solved with the AMD software update for windows.
Thanks for your help and sorry to have implicated darktable.

![2023-06-19 21_36_54-AMD Software_ Adrenalin Edition](https://github.com/darktable-org/darktable/assets/120166629/515636e4-70c9-4db2-880a-e2539fcddce2)

Greetings from Brussels,
Christian
